src/charm.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module defining the Charmed operator for FINOS Waltz."""
+
+import logging
+
+from ops import charm, main, model, pebble
+
+logger = logging.getLogger(__name__)
+
+
+class WaltzOperatorCharm(charm.CharmBase):
+    """A Juju Charm to deploy FINOS Waltz on Kubernetes.
+
+    This charm can be configured with postgresql database connection details which
+    will be used by FINOS Waltz.
+    """
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # General hooks:
+        self.framework.observe(self.on.waltz_pebble_ready, self._on_waltz_pebble_ready)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
```

Comment:
A minor nit: adding the event handlers immediately after the constructor (in the order they're subscribed to) can make for easier reading :)

Reply:
Done